spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/inbound/HazelcastClusterMonitorMessageProducer.java

```diff
@@ -0,0 +1,254 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.inbound;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.hazelcast.ClusterMonitorType;
+import org.springframework.integration.hazelcast.HazelcastIntegrationDefinitionValidator;
+import org.springframework.util.Assert;
+
+import reactor.util.StringUtils;
+
+import com.hazelcast.core.Client;
+import com.hazelcast.core.ClientListener;
+import com.hazelcast.core.DistributedObjectEvent;
+import com.hazelcast.core.DistributedObjectListener;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.LifecycleEvent;
+import com.hazelcast.core.LifecycleListener;
+import com.hazelcast.core.MemberAttributeEvent;
+import com.hazelcast.core.MembershipEvent;
+import com.hazelcast.core.MembershipListener;
+import com.hazelcast.core.MigrationEvent;
+import com.hazelcast.core.MigrationListener;
+
+/**
+ * Hazelcast Cluster Monitor Event Driven Message Producer is a message producer which
+ * enables {@link HazelcastClusterMonitorMessageProducer.HazelcastMembershipListener},
+ * {@link HazelcastClusterMonitorMessageProducer.HazelcastDistributedObjectListener}
+ * {@link HazelcastClusterMonitorMessageProducer.HazelcastMigrationListener}
+ * {@link HazelcastClusterMonitorMessageProducer.HazelcastLifecycleListener} and
+ * {@link HazelcastClusterMonitorMessageProducer.HazelcastClientListener} listeners in
+ * order to listen cluster related events and sends events to related channel.
+ *
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ */
+public class HazelcastClusterMonitorMessageProducer extends MessageProducerSupport {
+
+	private static final int CLUSTER_MONITOR_TYPE_COUNT = 5;
+
+	private final HazelcastInstance hazelcastInstance;
+
+	private Set<String> monitorTypes = Collections
+			.singleton(ClusterMonitorType.MEMBERSHIP.name());
+
+	private final Map<ClusterMonitorType, String> hazelcastRegisteredListenerIdMap = new ConcurrentHashMap<ClusterMonitorType, String>(
+			CLUSTER_MONITOR_TYPE_COUNT);
+
+	public HazelcastClusterMonitorMessageProducer(
+			final HazelcastInstance hazelcastInstance) {
+		Assert.notNull(hazelcastInstance, "'hazelcastInstance' must not be null");
+		this.hazelcastInstance = hazelcastInstance;
+	}
+
+	public void setMonitorEventTypes(String monitorEventTypes) {
+		HazelcastIntegrationDefinitionValidator.validateEnumType(
+				ClusterMonitorType.class, monitorEventTypes);
+		final Set<String> monitorTypes = StringUtils
+				.commaDelimitedListToSet(monitorEventTypes);
```

Comment:
I think `validate...` method can return the `Set<ClusterMonitorType>` already avoiding double `commaDelimitedListToSet` call. 
In addition I think we should add `vararg` methods for the Java Config convenience.

But that should be done in the separate JIRA.